server/app.go

```diff
@@ -33,10 +33,11 @@ func (a *App) Run(addr string) {
 	log.Fatal(http.ListenAndServe(addr, a.Router))
 }
 
-func (a *App) getTimeTable(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintf(w, "Time table")
+func (a *App) initializeRoutes() {
+	a.Router.HandleFunc("/segments", a.createSegmentMiddleware).Methods("POST")
 }
 
-func (a *App) initializeRoutes() {
-	a.Router.HandleFunc("/results", a.getTimeTable).Methods("GET")
+func (a *App) createSegmentMiddleware(w http.ResponseWriter, r *http.Request) {
```

Comment:
I don't think this is a middleware. This is a simple handler.

Reply:
Actually the whole point of this is just to pass the db object as well to the `createSegment` function. :trollface: If there would be some better way for this, I could eliminate these completely. 🤔

Reply:
Check out the blog post I linked and look for "Handlers hang off the server". I think that solves this